src/main/scala/com/metamx/tranquility/beam/ClusteredBeam.scala

```diff
@@ -214,18 +257,40 @@ class ClusteredBeam[EventType: Timestamper, InnerBeamType <: Beam[EventType]](
               assert(prevBeamDicts.size < tuning.partitions)
               assert(timestamp > prev.latestCloseTime)
 
-              // We might want to cover multiple time segments in advance.
-              val numSegmentsToCover = tuning.minSegmentsPerBeam +
-                rand.nextInt(tuning.maxSegmentsPerBeam - tuning.minSegmentsPerBeam + 1)
-              val intervalToCover = timestamp to tuning.segmentGranularity.increment(timestamp, numSegmentsToCover)
-              val timestampsToCover = tuning.segmentGranularity.getIterable(intervalToCover).asScala.map(_.start)
+              var optionalIntervalToCover = None: Option[Interval]
+              var optionalTimestampsToCover = None: Option[Iterable[DateTime]]
+              if(allowGranularityChange) {
```

Comment:
we will never create overlapping segments here - may be I did not understand your concern.